tests/test_autoingest.py

```diff
@@ -0,0 +1,270 @@
+        """test ingesting an ComCam file
+        """
+        fits_name = "3019053000001-R22-S00-det000.fits.fz"
+        config = self.createConfig("ingest_comcam_gen3.yaml", fits_name)
+
+        # setup directory to scan for files in the forwarder staging directory
+        # and ensure one file is there
+        ingesterConfig = config["ingester"]
+        forwarder_staging_dir = ingesterConfig["forwarderStagingDirectory"]
+        scanner = DirectoryScanner([forwarder_staging_dir])
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 1)
+
+        # create the file ingester, get all tasks associated with it, and
+        # create the tasks
+        ingester = FileIngester(ingesterConfig)
+        clean_tasks = ingester.getButlerCleanTasks()
+
+        task_list = []
+        for clean_task in clean_tasks:
+            task = asyncio.create_task(clean_task())
+            task_list.append(task)
+
+        # check to see that the file is there before ingestion
+        self.assertTrue(os.path.exists(self.destFile))
+        print(f"destFile = {self.destFile}")
+
+        await ingester.ingest([self.destFile])
+
+        # make sure forwarder staging area is now empty
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 0)
+
+        # Check to see that the file was ingested.
+        # Recall that files start in the forwarder staging area, and are
+        # moved to the butler staging area before ingestion. On "direct"
+        # ingestion, this is where the file is located.  This is a check
+        # to be sure that happened.
+        name = self.strip_prefix(self.destFile, self.forwarderDir)
+        file_to_ingest = os.path.join(self.stagingDir, name)
+        self.assertTrue(os.path.exists(file_to_ingest))
+
+        # this file should now not exist
+        self.assertFalse(os.path.exists(self.destFile))
+
+        # add one more task, whose sole purpose is to interrupt the others by
+        # throwing an acception
+        task_list.append(asyncio.create_task(self.interrupt_me()))
+
+        # kick off all the tasks, until one (the "interrupt_me" task)
+        # throws an exception
+        try:
+            await asyncio.gather(*task_list)
+        except Exception:
+            for task in task_list:
+                task.cancel()
+
+        # that should have been enough time to run the "real" tasks,
+        # which performed the ingestion, and the clean up task, which
+        # was set to clean it up right away.  (That "clean up" time
+        # is set in the config file loaded for this FileIngester).
+        # And, when "cleaned up", the file that was originally there
+        # is now gone.  Check for that.
+        self.assertFalse(os.path.exists(file_to_ingest))
+
+        # check to be sure that the file wasn't "bad" (and
+        # therefore, not ingested)
+        bad_path = os.path.join(self.badDir, fits_name)
+        self.assertFalse(os.path.exists(bad_path))
```

Comment:
To be honest I don't quite understand why separate tests for ingesting an image from different instruments are necessary (and why they are so different from each other). Are Auxtel data handled differently by **ctrl_oods** comparing to Comcam data? Or do these tests test different functionalities and using different instruments is sort of a red herring?

Reply:
I wanted to test the results on data specific to each instrument to be sure there were no underlying problems that caused issues with one or the other.